administrator/components/com_patchtester/PatchTester/Model/PullModel.php

```diff
@@ -218,24 +220,40 @@ private function applyWithCIServer($id)
 			return false;
 		}
 
-		// Check if zip folder exists on server
-		$serverHeaders = @get_headers($serverZipPath);
+		$version    = new Version;
+		$httpOption = new Registry;
+		$httpOption->set('userAgent', $version->getUserAgent('Joomla', true, false));
 
-		if (!$serverHeaders || $serverHeaders[0] != 'HTTP/1.1 200 OK')
+		// Try to download the zip file
+		try
+		{
+			$http = HttpFactory::getHttp($httpOption);
+			$result = $http->get($serverZipPath);
+		}
+		catch (\RuntimeException $e)
+		{
+			$result = null;
+		}
+
+		if ($result === null || ($result->code != 200 && $result->code != 310))
```

Comment:
@richard67 Please use the functions instead of the magic getter. So `$result->getStatusCode()`